### PR TITLE
WT-11530 Package common variant variables in evergreen file into expansions

### PR DESCRIPTION
**Pull request by jiechenbo**

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -68,6 +68,11 @@ functions:
               export UBSAN_OPTIONS="$COMMON_SAN_OPTIONS:print_stacktrace=1"
             fi
 
+            if [ -n "${num_threads}" ]; then
+              num_threads="${num_threads}"
```
Comment (mickgraham-mongodb) [inline test/evergreen.yml:72]: Its not really the number of threads, but number parallel or number of processes.

File: test/evergreen.yml [hunk 2]
```diff
@@ -158,8 +170,9 @@ functions:
         # Note that the config flags are resolved prior to changing to cmake_build directory.
         DEFINED_EVERGREEN_CONFIG_FLAGS="${CMAKE_BUILD_TYPE|} \
           ${CMAKE_INSTALL_PREFIX|-DCMAKE_INSTALL_PREFIX=$(pwd)/cmake_build/LOCAL_INSTALL} \
-          ${CMAKE_PREFIX_PATH|} \
+          ${CMAKE_PREFIX_PATH|-DCMAKE_PREFIX_PATH="$(pwd)/TCMALLOC_LIB"} \
           ${CMAKE_TOOLCHAIN_FILE|-DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_stable_gcc.cmake} \
+          ${ENABLE_TCMALLOC|-DENABLE_TCMALLOC=1}
```
Comment (mickgraham-mongodb) [inline test/evergreen.yml:175]: I like this to make it explicit.

File: test/evergreen.yml (hunk 1 above)
Comment (jiechenbo) [reply test/evergreen.yml]: Done, i renamed it

File: test/evergreen.yml (hunk 2 above)
Comment (jiechenbo) [reply test/evergreen.yml]: I have removed the default value, and made sure that each variant has ENABLE_TCMALLOC=0 or ENABLE_TCMALLOC=1

File: test/evergreen.yml [hunk 3]
```diff
@@ -496,8 +498,8 @@ functions:
         ${PREPARE_TEST_ENV}
         . test/evergreen/find_cmake.sh
         cd cmake_build
-        echo "Using smp_command '${smp_command}' for 'make check all'"
-        $CTEST -L check ${smp_command|} --output-on-failure ${check_args|} 2>&1
+        echo "Using num threads '${num_parallel}' for 'make check all'"
```
Comment (mickgraham-mongodb) [inline test/evergreen.yml:501]: Number of parallel instead of threads?

File: test/evergreen_develop.yml [hunk 4]
```diff
@@ -11,10 +11,8 @@ variables:
     additional_env_vars: export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$WT_BUILDDIR/test/utility/"
     CC_OPTIMIZE_LEVEL: -DCC_OPTIMIZE_LEVEL=-O3
     HAVE_DIAGNOSTIC: -DHAVE_DIAGNOSTIC=0
-    CMAKE_TOOLCHAIN_FILE: -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_stable_gcc.cmake
-    CMAKE_INSTALL_PREFIX: -DCMAKE_INSTALL_PREFIX=$(pwd)/cmake_build/LOCAL_INSTALL
+    ENABLE_TCMALLOC: -DENABLE_TCMALLOC=0
     smp_command: -j $(echo "`grep -c ^processor /proc/cpuinfo` * 2" | bc)
```
Comment (mickgraham-mongodb) [inline test/evergreen_develop.yml:15]: Should this be num_parallel?

File: test/evergreen_develop.yml [hunk 5]
```diff
@@ -157,9 +158,7 @@ buildvariants:
   run_on:
   - ubuntu2004-test
   expansions:
-    CMAKE_PREFIX_PATH: -DCMAKE_PREFIX_PATH="$(pwd)/TCMALLOC_LIB"
     smp_command: -j $(echo "`grep -c ^processor /proc/cpuinfo` * 2" | bc)
```
[the remark this answers is not in the document]
Comment (jiechenbo) [reply test/evergreen_develop.yml]: i missed it, thanks corrected now

File: test/evergreen_develop.yml (hunk 4 above)
Comment (jiechenbo) [reply test/evergreen_develop.yml]: i missed it, thanks corrected now

File: test/evergreen.yml (hunk 3 above)
Comment (jiechenbo) [reply test/evergreen.yml]: changed! thank you

**Review by mickgraham-mongodb — APPROVED** [review]
LGTM

File: test/evergreen.yml [hunk 6]
```diff
@@ -78,6 +78,7 @@ functions:
               export PATH=/opt/mongodbtoolchain/v4/bin:$PATH
             fi
 
+
```
Comment (etienneptl) [inline test/evergreen.yml:81]: Extra line

File: test/evergreen.yml [hunk 7]
```diff
@@ -272,9 +274,9 @@ functions:
           # Compiling with CMake generated Ninja file.
           cd cmake_build
           if [ "${cmake_generator|Ninja}" = "Ninja" ]; then
-            ninja ${smp_command|} 2>&1
+            ninja ${num_parallel} 2>&1
```
Comment (etienneptl) [inline test/evergreen.yml:277]: As discussed, I think this is confusing as from the variable name, you don't expect the `-j` to be included. The `smp_command` was probably a good name for that reason.

**Review by etienneptl — APPROVED** [review]
LGTM, love those improvements! I have on np about the `-j` arg.

**Review by etienneptl — APPROVED** [review]
LGTM!

**Comment by jiechenbo** [comment]
evergreen merge

**Pull request merged**